indico/modules/events/management/client/js/EmailAuthors.jsx

```diff
@@ -0,0 +1,211 @@
+
+  useEffect(() => {
+    if (!triggerSelector) {
+      return;
+    }
+    const handler = () => setOpen(true);
+    const element = document.querySelector(triggerSelector);
+    element.addEventListener('click', handler);
+    return () => element.removeEventListener('click', handler);
+  }, [triggerSelector]);
+
+  return (
+    open && (
+      <EmailAuthorsModal
+        eventId={eventId}
+        sourceIds={sourceIds}
+        context={context}
+        onClose={() => setOpen(false)}
+      />
+    )
+  );
+}
+
+EmailAuthorsButton.propTypes = {
+  eventId: PropTypes.number.isRequired,
+  context: PropTypes.oneOf(Object.keys(CONTEXTS)).isRequired,
+  paramsSelector: PropTypes.string.isRequired,
+  triggerSelector: PropTypes.string.isRequired,
+};
+
+export function EmailAuthorsModal({eventId, sourceIds, context, onClose}) {
+  const [selectedPersons, setSelectedPersons] = useState([]);
+  const [selectedUsers, setSelectedUsers] = useState([]);
+  const [successMessage, setSuccessMessage] = useState(undefined);
+  const [activeFilters, setActiveFilters] = useState([]);
+  const formData = new FormData();
+  sourceIds.forEach(v => formData.append(`${context}_id`, v));
+  const {data, loading} = useIndicoAxios(
+    {
+      url: CONTEXTS[context].personListURL({event_id: eventId}),
+      method: 'POST',
+      data: formData,
+      headers: {'content-type': 'multipart/form-data'},
+    },
+    {camelize: true}
+  );
+  const extraRoles =
+    context === 'abstract'
+      ? [
+          {
+            icon: 'user-check icon',
+            isActive: p => p.submitter,
+            titleActive: Translate.string('This person submitted an abstract'),
+            titleInactive: Translate.string("This person hasn't submitted any abstracts"),
+          },
+        ]
+      : undefined;
+
+  const onChangeSelection = ({person, user}) => {
+    person && setSelectedPersons(person);
+    user && setSelectedUsers(user);
+  };
+
+  const isVisible = person =>
+    !activeFilters.length || activeFilters.every(filter => FILTER_OPTIONS[filter].isMatch(person));
+
+  const toggleFilter = name => {
+    setActiveFilters(
+      activeFilters.includes(name)
+        ? activeFilters.filter(f => f !== name)
+        : [...activeFilters, name]
+    );
+  };
+
+  const filterButtons = Object.entries(FILTER_OPTIONS)
+    .filter(([, filter]) => !filter.contexts || filter.contexts.includes(context))
+    .map(([name, filter]) => ({
+      key: name,
+      content: filter.text,
+      onClick: () => toggleFilter(name),
+      active: activeFilters.includes(name),
+    }));
+
+  return (
+    <Modal open onClose={() => onClose()} size="large">
+      <Modal.Header content={Translate.string('Email authors')} />
+      <Modal.Content>
+        {successMessage}
+        <div>
+          <Translate
+            as="div"
+            style={{display: 'inline-block', fontWeight: 'bold', padding: '0.5em'}}
+          >
+            Filter by
+          </Translate>
```

Comment:
SUI doesn't have something nicer to attach a label to a button group? :/

Reply:
I don't think so... We could use a disabled button I guess but it would still need custom CSS

Reply:
not a big deal - i don't think the current style is particularly bad ;)